lib/services/api/api_service.dart

```diff
@@ -0,0 +1,74 @@
+import 'package:survey/gen/configs.gen.dart';
+import 'package:survey/services/http/http_service.dart';
+import 'package:object_mapper/object_mapper.dart';
+import 'package:survey/services/locator/locator_service.dart';
+
+part 'api_exception.dart';
+
+part 'api_params.dart';
+part 'api_service_register.dart';
+
+abstract class ApiService {
+  Future<T> call<T extends Mappable>({
+    required HttpMethod method,
+    String? baseUrl,
+    required String endPoint,
+    ApiParams? params,
+    String? token,
+  });
+
+  void configureGlobalBaseUrl(String? baseUrl);
+
+  void configureGlobalToken(String? token);
+}
+
+class ApiServiceImpl implements ApiService {
+  static String? _fallbackBaseUrl;
+  static String? _fallbackToken;
+  final HttpService _httpService = locator.get();
+
+  @override
+  Future<T> call<T extends Mappable>({
+    required HttpMethod method,
+    String? baseUrl,
+    required String endPoint,
+    ApiParams? params,
+    String? token,
+  }) async {
+    final String? finalBaseUrl = baseUrl ?? _fallbackBaseUrl;
+    assert(finalBaseUrl != null);
+
+    final Map<String, dynamic> headers = {};
+
+    // Append bearer token
+    final String? finalToken = token ?? _fallbackToken;
+    if (finalToken != null) {
+      headers["authorization"] = "Bearer $finalToken";
```

Comment:
assuming Bearer is a static value always could be a bit troublesome later, usually you will receive the token-type and access-token from a response.